File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/basic/BasicDateMessage.scala





















// Generated on 05/03/2013 10:34:35
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait BasicDateMessage extends MessageDeserializer {
  val messageId = 177

  protected trait Message extends BaseMessage with Serializable {
    val messageId = BasicDateMessage.messageId

    val day: Byte
    val month: Byte
    val year: Short

    def serialize(buf: Buffer) {
      buf.writeByte(day)
      buf.writeByte(month)
      buf.writeShort(year)
    }

  }

  protected def create(day: Byte, month: Byte, year: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val day = buf.readByte()
    val month = buf.readByte()
    val year = buf.readShort()

    create(day, month, year)
  }

}

object BasicDateMessage extends BasicDateMessage {
  case class MessageImpl(day: Byte, month: Byte, year: Short) extends Message
  type Target = MessageImpl

  override def create(day: Byte, month: Byte, year: Short) = MessageImpl(day, month, year)

  def apply(day: Byte, month: Byte, year: Short) = create(day, month, year)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/BasicDateMessage.java


// Generated on 05/08/2013 19:37:41
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class BasicDateMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 177;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte day;
    public byte month;
    public short year;
    
    public BasicDateMessage() { }
    
    public BasicDateMessage(byte day, byte month, short year) {
        this.day = day;
        this.month = month;
        this.year = year;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(day);
        buf.writeByte(month);
        buf.writeShort(year);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        day = buf.readByte();
        if (day < 0)
            throw new RuntimeException("Forbidden value on day = " + day + ", it doesn't respect the following condition : day < 0");
        month = buf.readByte();
        if (month < 0)
            throw new RuntimeException("Forbidden value on month = " + month + ", it doesn't respect the following condition : month < 0");
        year = buf.readShort();
        if (year < 0)
            throw new RuntimeException("Forbidden value on year = " + year + ", it doesn't respect the following condition : year < 0");
    }
    
}
